spring-cloud-starter-stream-processor-httpclient/README.adoc

```diff
@@ -3,6 +3,34 @@
 
 A processor app that makes requests to an HTTP resource and emits the response body as a message payload. This processor can be combined, e.g., with a time source app to periodically poll results from a HTTP resource.
 
+== Input
+=== Headers
+Any Required HTTP headers must be explicitly set via the `headers-expression` property. See examples below.
+Header values may also be used to construct the request body when referenced in the `body-expression` property.
+
+=== Payload
+The Message payload may be any Java type. Generally, standard Java types such as String(e.g., JSON, XML) or byte array
+payloads are recommended. A Map should work without too much effort. By default, the payload will become HTTP request
+body (if needed). You may also set the `body-expression` property to construct a value derived from the Message, or
+`body` to use a static (literal) value.
+
+Internally, the processor uses
+https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html#exchange-org.springframework.http.RequestEntity-java.lang.Class-[RestTemplate.exchange(...)].
+
+The RestTemplate supports Jackson JSON serialization to support any request and response types if necessary. The
+`expected-response-type` property (`String.class` by default). )may be set to any class in your application
```

Comment:
I think this is a typo: 
`s/)/It` may be...

Otherwise LGTM, but as I said in other places I'm not sure that we all do the proper, expected work here for inputs/outputs...

/CC @viniciusccarvalho , @sabbyanandan